File: Strings/Anagram/Anagram.java
import java.util.*;

public class Anagram {

	static HashMap<Integer, Integer> s1 = new HashMap<Integer, Integer

	static int anagraming(String subject) {
		int totalLength = subject.length();
		int firstLength = (int) totalLength / 2;
		int secondLength = totalLength - firstLength;
		if (firstLength != secondLength) {
			return -1;
		}
		for (int i = 0; i < firstLength; i += 1) {
			if (subject.charAt(i) != subject.charAt(totalLength-1-i)){
				count += 1;
			}
		}
		return count;
	}

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int t = in.nextInt();
		in.nextLine();
		for (int i = 0; i < t; i += 1) {
			String toBeAnagramed = in.nextLine();
			System.out.println(anagraming(toBeAnagramed));
		}
	}
}

File: Strings/Anagram/Anagram.py

first_words = {}
second_words = {}



def initialize_dict() :
	starter = ord('a')
	while(starter <= ord('z')):
		first_words[chr(starter)] = 0
		second_words[chr(starter)] = 0
		starter += 1

def anagramed(word):
	total_length = len(word)
	first_length = total_length // 2
	second_length = total_length - first_length
	if (first_length != second_length):
		return -1

	for index in range(first_length):
		first_words[word[index]] += 1
		second_words[word[total_length - 1 - index]] += 1
	
	count = 0
	for key in second_words:
		count += abs(first_words[key] - second_words[key])
	return count // 2

t = int(input())
for n in range(t):
	to_be_anagramed = input()
	initialize_dict()
	print(anagramed(to_be_anagramed))



